### DISCUSSION TITLE
[Pixie] TrafficNet & Protocol Inference Models

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#587] [Google Summer of Code 2022]
Pixie’s protocol tracer captures traffic for various protocols, including HTTP, gRPC, postgres and many more. The protocol tracer relies on protocol inference rules to classify traffic based on the contents of the messages being sent. To validate the accuracy of our protocol inference rules, we need a large database of different traffic patterns. We call this dataset TrafficNet. For this project, you will try to (1) expand the TrafficNet data set to include more samples of existing and new protocols, and (2) experiment with different inference models to improve the accuracy of the protocol inference.

Mentor(s): Omid Azizi (@oazizi000)

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#trafficnet--protocol-inference-models

**Comment by mory91** [comment]
I would be happy to work on this project. Can you provide some details and resources to look for, for a better proposal?

**Reply by nate-double-u** [reply]
Hi, it looks like someone's answered your question in the upstream issue 🙂

https://github.com/pixie-io/pixie/issues/404#issuecomment-1076760698
